data_structures/arrays/strings/almost_palindrome_checker.rb

```diff
@@ -0,0 +1,59 @@
+# Challenge name: Almost Palindrome
+#
+# Given a string s, return true if the s can be palindrome after deleting at most one character from it.
+#
+# Example 1:
+# Input: s = "aba"
+# Output: true
+#
+# Example 2:
+# Input: s = "abca"
+# Output: true
+# Explanation: You could delete the character 'c'.
+#
+# Example 3:
+# Input: s = "abc"
+# Output: false
+#
+# Constraints:
+# 1 <= s.length <= 105
+# s consists of lowercase English letters.
+
+# Approach 1: Two Pointers
+# Time Complexity: O(n)
+
+def almost_palindrome_checker(string)
+  p1 = 0
+  p2 = string.length - 1
```

Comment:
I think I'd move this `split` method outside the while just to reduce the number of operations. You're splitting the array multiple times and it's not really needed

```suggestion
  p1 = 0
  p2 = string.length - 1
  break_string = string.split('')
```